File: Exercism Codes/R/darts.R
score <- function(x, y) {
  dst <- sqrt(x * x + y * y)
  if(dst > 10) {
    return(0)
  }
  else if(dst > 5) {
    return(1)
  }
  else if(dst > 1) {
    return(5)
  }
  else {
    return(10)
  }
}

File: Exercism Codes/Kotlin/Darts.kt
import kotlin.math.sqrt

object Darts {

    fun <T: Number> score(x: T, y: T): Int {
        val d = sqrt(x.toDouble() * x.toDouble() + y.toDouble() * y.toDouble())
        return when {
            d > 10 -> 0
            d > 5 -> 1
            d > 1 -> 5
            else -> 10
        }
    }
}
